src/cli.rs

```diff
@@ -3524,5 +3542,83 @@ mod tests_aoflagger_flagset {
         args.extend_from_slice(&gpufits_paths);
 
         BirliContext::from_args(&args).unwrap();
+
+        let birli_ctx = BirliContext::from_args(&args).unwrap();
+
+        assert!(birli_ctx.prep_ctx.correct_cable_lengths);
+        assert!(matches!(birli_ctx.prep_ctx.passband_gains, None));
+        assert!(birli_ctx.prep_ctx.correct_digital_gains);
+        assert!(birli_ctx.prep_ctx.correct_geometry);
+        assert!(matches!(birli_ctx.prep_ctx.aoflagger_strategy, Some(_)));
+        assert_eq!(
+            birli_ctx.io_ctx.metafits_in.display().to_string(),
+            metafits_path
+        );
+        assert_eq!(
+            birli_ctx.io_ctx.flag_template,
+            Some(mwaf_path_template.to_str().unwrap().into())
+        );
+
+        let corr_ctx = CorrelatorContext::new(
+            birli_ctx.io_ctx.metafits_in.clone(),
+            &birli_ctx.io_ctx.gpufits_in,
+        )
+        .unwrap();
+
+        birli_ctx.run().unwrap();
+
+        let gpubox_ids: Vec<usize> = corr_ctx
+            .common_coarse_chan_indices
+            .iter()
+            .map(|&chan| corr_ctx.coarse_chans[chan].gpubox_number)
+            .collect();
+
+        assert!(!gpubox_ids.is_empty());
+
+        let flag_file_set = FlagFileSet::open(
+            mwaf_path_template.to_str().unwrap(),
+            &gpubox_ids,
+            corr_ctx.mwa_version,
+        )
+        .unwrap();
+
+        // If we have the right version of AOFlagger, check the output flags.
```

Comment:
@d3v-null This is the test I wrote for chunked flags. What do you think?

Reply:
only issues here is that the test doesn't actually fail if the version or strategy file is wrong (if i'm understanding the test correctly)